### ISSUE TITLE
Component Outline

### ISSUE THREAD

**Issue by cmaycumber** [nandorojo/dripsy#8]
Here's a non-comprehensive outline of potential components that could be included in Dripsy: 

**HTML Elements** - @expo/html-elements
- H1
- H2
- H3
- H4
- H5
- H6
- H6
- A
- P
- Others?

**React Native**
- View
- ScrollView
- FlatList
- Text
- SafeAreaView
- ActivityIndicator
- ImageBackground
- Image
- KeyboardAvoidingView
- Picker
- Pressable - Needs RN 63.0
- SectionList
- StatusBar
- TouchableHighlight
- TouchableOpacity
- TouchableWithoutFeedback
- VirtualizedList
- Do we want the entire library or only specific ones?

**Theme-UI Inspired**
- Flex
- Checkbox
- Slider
- Switch
- Text
- Heading
- Badge
- Avatar
- Divider
- Progress
- Link
- Container
- Select
- Textarea
- TextInput
- SVG

**Additional**
- Segmented Control
- Modal
- Additional Components TBD

**Potential Questions**
- How much should we rely on outside dependencies? (e.i @react-native-community/checkbox, etc.) A lot might be included in Expo by default.
- Can we add a lot of components without making it too opinionated?
- Should we have heading + HTML elements, Link + A, or consolidate them somehow? 

Note that this list is non-comprehensive and any feedback on to what it should be added/removed is welcomed to improve the list.

**Comment by nandorojo** [comment]
Thanks for putting this together. A few ideas:

For components like `TextInput`, Dripsy could follow in the steps of `theme-ui` and add some base level styles that make it less ugly than the standard component.

A number of these are already exported, as seen here: https://github.com/nandorojo/dripsy/blob/master/src/components/index.tsx

Most of them have no added styles, however.

It could be smart to also have a checkbox for them being documented, but this probably makes more sense once they're all actually exported by Dripsy.

**Comment by cmaycumber** [comment]
Thanks for the feedback! I think the base level styles are a good idea. 

For the current ones that are exported in terms of file structure do you think they should continue being exported directly from the index file? I was playing around with the idea of an html.tsx file and a react-native.tsx file but wasn't 100% sure. 

Another note in terms of Heading and Text in theme-ui. We may somehow be able to consolidate the HTML components into some of the theme-ui components as variants or a prop but I can also see the case for exporting them as completely separate components.

**Comment by nandorojo** [comment]
> Do we want the entire library or only specific ones?

Good question. I think having most components that are styled commonly and don't have strong base styles should be exported. For example, `Button` shouldn't be.

It could be cool if components that receive a color prop, such as `ActivityIndicator`, could pass theme color values directly.

**Comment by nandorojo** [comment]
> For the current ones that are exported in terms of file structure do you think they should continue being exported directly from the index file? I was playing around with the idea of an html.tsx file and a react-native.tsx file but wasn't 100% sure.

Both make sense. For my own projects, I definitely split code up more, but I like the idea of fewer files for open source. It makes it easier for people to contribute and see how the code is structured. So for now, unless there's a strong counter point, I'd encourage keeping the component exports in one file. I'm open to this changing down the line, though.

**Comment by nandorojo** [comment]
> Another note in terms of Heading and Text in theme-ui. We may somehow be able to consolidate the HTML components into some of the theme-ui components as variants or a prop but I can also see the case for exporting them as completely separate components.

Yeah, good point. It seems like the HTML approach would look like the `Styled.p`, `Styled.h1` approach from `theme-ui`. 

I personally like the pattern of having universal `H1`, `H2`, `P` components, which can then be edited in the theme with a pattern like this:

```js
export default {
  text: {
    p: { fontWeight: 'bold', }
    h1: { marginBottom: 2 }
  }
}
```

It could just be what I'm used to, but I tend to like separating each UI element into its own JSX component, especially since RN doesn't have string elements the way HTML does. Worst case, people can always use the `as` prop on any component.

**Comment by cmaycumber** [comment]
> Good question. I think having most components that are styled commonly and don't have strong base styles should be exported. For example, Button shouldn't be.
> 
> It could be cool if components that receive a color prop, such as ActivityIndicator, could pass theme color values directly. Not a priority though.

I'm in 100% agreement with this. That makes the most sense to me as well. 

> I personally like the pattern of having universal H1, H2, P components, which can then be edited in the theme with a pattern like this:

I like this pattern better. Styled was one of the features I personally didn't really find myself using.

**Comment by cmaycumber** [comment]
I've started to add a few of these in. I'll create a pull request to merge in some of them then we can add/remove as we please.

**Comment by daxfrost** [comment]
Hi @cmaycumber, I am trying to use the H1 component that already seems to be in the latest release...

Is it expected to be usable? Unfortunately I can't find any example usages of it.

TS is throwing an error:

![image](https://user-images.githubusercontent.com/1788177/94045412-6be3b780-fd9d-11ea-893a-04fd14f38edb.png)

I am also getting an issue when actually rendering in RN, hopefully I am also just using it incorrectly:

![image](https://user-images.githubusercontent.com/1788177/94045577-a188a080-fd9d-11ea-9ef0-d02935d13dc6.png)

**Comment by nandorojo** [comment]
I've seen this TS error too, but it actually does still work for me when I ignore the error. I'm on the `fresnel-2` branch, though.

**Comment by daxfrost** [comment]
Yeah, it does still work for me if I ignore the error.

**Comment by nandorojo** [comment]
TS errors are fixed in `1.2.3`.
